python/ffsim/qiskit/transpiler_passes/lucj_heavy_hex_preset_pass_manager.py

```diff
@@ -0,0 +1,298 @@
+    the routine ends and returns the `graph_new`. If not, it removes an alpha-beta
+    interaction pair from the end of list `alpha_beta_indices` and checks for
+    subgraph isomorphism again. It cycle continues, until a isomorhic subgraph is
+    found.
+
+    Args:
+        num_orbitals: Number of orbitals, i.e., number of nodes in each alpha-alpha
+            linear chain.
+        backend_coupling_graph: The coupling graph of the backend on which the LUCJ
+            ansatz will be mapped and run. This function takes the coupling graph as
+            a undirected `rustworkx.PyGraph` where there is only one 'undirected' edge
+            between two nodes, i.e., qubits. Usually, the coupling graph of a IBM
+            backend is directed (e.g., Eagle devices such as `ibm_brisbane`) or may
+            have two edges between same two nodes (e.g., Heron `ibm_torino`). This
+            function is only compatible with undirected graphs where there is only
+            a single undirected edge between same two nodes.
+
+    Returns:
+        graph_new: A graph that has the _zigzag_ pattern and is an isomorphic subgraph
+            to an a heavy-hex IBM backend.
+        num_alpha_beta_qubits: Number of connecting qubits between the linear chains
+            in the zigzag pattern. While we want as many connecting (alpha-beta) qubits
+            between the linear (alpha-alpha) chains, we cannot accomodate all due to
+            connectivity constraints of backends. This is the maximum number of
+            connecting qubits the zigzag pattern can have while being backend compliant
+            (i.e., isomorphic subgraph to the backend coupling graph).
+    """
+    isomorphic = False
+    graph = _create_two_linear_chains(num_orbitals=num_orbitals)
+
+    graph_new = copy.deepcopy(graph)  # to avoid not bound warning
+    while not isomorphic:
+        graph_new = copy.deepcopy(graph)
+
+        if not alpha_beta_indices:
+            break
+
+        # add new nodes and edges
+        for i, (a, b) in enumerate(sorted(alpha_beta_indices, key=lambda x: x[0])):
+            new_node = 2 * num_orbitals + i
+            graph_new.add_node(new_node)
+            graph_new.add_edge(a, new_node, None)
+            graph_new.add_edge(new_node, b + num_orbitals, None)
+
+        isomorphic = rustworkx.is_subgraph_isomorphic(
+            backend_coupling_graph,
+            graph_new,
+            call_limit=1_000_000,
+            id_order=False,
+            induced=False,
+        )
+
+        if not isomorphic:
+            warnings.warn(
+                f"Backend cannot accomodate alpha_beta_incides {alpha_beta_indices}.\n "
+                f"Removing interaction {alpha_beta_indices[-1]} from the end."
+            )
+            del alpha_beta_indices[-1]
+
+    return graph_new, alpha_beta_indices
+
+
+def _make_backend_cmap_pygraph(backend: BackendV2) -> PyGraph:
```

Comment:
Given that this function only uses `backend.coupling_map.graph` from the backend, I think it should be refactored so that it simply accepts that graph directly as input.